django/template/loader_tags.py

```diff
@@ -207,6 +210,11 @@ def render(self, context):
         except Exception:
             if context.template.engine.debug:
                 raise
+            else:
+                template_name = getattr(context, 'template_name', 'unknown')
+                logger.warn('Exception raised while rendering {{% include %}} for '
```

Comment:
would you like this change applied to the other new template logger as well?
On Sep 18, 2015 5:00 PM, "Tim Graham" <notifications@github.com> wrote:

> In django/template/loader_tags.py
> <https://github.com/django/django/pull/5285#discussion_r39912105>:
>
> > @@ -207,6 +210,11 @@ def render(self, context):
> >          except Exception:
> >              if context.template.engine.debug:
> >                  raise
> > +            else:
> > +                template_name = getattr(context, 'template_name', 'unknown')
> > +                logger.warn('Exception raised while rendering {{% include %}} for '
>
> Instead of interpolating the parameter in the message, logging should do
> that work, so this should be:
>
> logger.warn(
>     'Exception raised while rendering {%% include %%} for '
>     'template %s. Empty string rendered instead.',
>     template_name,
>     exc_info=True,
> )
>
> I am not sure if template name retrieval works in the 'unknown' case
> (seems to return "None" in the test case). Would be nice to have a test for
> the case when a template name does exist too.
> Edits: http://dpaste.com/1XTAAT0
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/django/django/pull/5285/files#r39912105>.
>